app/src/main/java/com/babylon/wallet/android/presentation/accessfactorsource/AccessFactorSourceBottomSheet.kt

```diff
@@ -36,15 +51,25 @@ fun AccessFactorSourceBottomSheet(
 
     AccessFactorSourceBottomSheetContent(
         modifier = modifier,
+        isAccessingFactorSourceInProgress = state.isAccessingFactorSourceInProgress,
+        isAccessingFactorSourceCompleted = state.isAccessingFactorSourceCompleted,
+        showContentForFactorSource = state.showContentFor,
         onDismiss = onDismiss
     )
 }
 
 @Composable
 private fun AccessFactorSourceBottomSheetContent(
     modifier: Modifier = Modifier,
+    isAccessingFactorSourceInProgress: Boolean,
+    isAccessingFactorSourceCompleted: Boolean,
+    showContentForFactorSource: AccessFactorSourceUiState.ShowContentFor,
```

Comment:
Currently this content is only for deriving public keys, correct? If so we should separate each case on its own composable or have different dialogs for different cases. Like

Instead of naming this `AccessFactorSourceBottomSheet` we could name it like `DerivePublicKeysDialog`. Then we can create other dialogs that do other things with factor sources and let the app event decide what dialog to open.

Reply:
that's also in my mind for a future change. But my answer here is similar to the above. Let's see how it will really escalate. Yeah it seems that one dialog might be too much to handle all the factor sources. 
But let's go step by step.

Reply:
Agreed